src/templates/product.js

```diff
@@ -19,15 +19,15 @@ const productTemplate = {
                           Left
                           <img class="carousel-button-arrow" src="//sdks.shopifycdn.com/buy-button/latest/arrow.svg" alt="Carousel Arrow"/>
                         </button>
-                        <img class="{{data.classes.product.img}}" src="{{data.currentImage.src}}" data-element="product.img" />
+                        <img class="{{data.classes.product.img}}" alt="{{data.currentImage.altText}}" src="{{data.currentImage.src}}" data-element="product.img" />
                         <button type="button" class="carousel-button carousel-button--next">
                           Right
                           <img class="carousel-button-arrow" src="//sdks.shopifycdn.com/buy-button/latest/arrow.svg" alt="Carousel Arrow"/>
                         </button>
                       </div>
                       <div class="{{data.classes.product.carousel}}">
                         {{#data.carouselImages}}
-                        <a data-element="product.carouselitem" href="{{src}}" class="{{data.classes.product.carouselItem}} {{#isSelected}} {{data.classes.product.carouselItemSelected}} {{/isSelected}}" data-image-id="{{id}}" style="background-image: url({{carouselSrc}})"></a>
+                        <a data-element="product.carouselitem" alt="{{altText}}" href="{{src}}" class="{{data.classes.product.carouselItem}} {{#isSelected}} {{data.classes.product.carouselItemSelected}} {{/isSelected}}" data-image-id="{{id}}" style="background-image: url({{carouselSrc}})"></a>
```

Comment:
The `alt` attribute doesn't have an impact when used on an `<a>` element.

I'm not sure why the code is using `background-image` instead of a real `<img>` element child. Maybe so the image won't load until that item is selected? But the original code is really, really bad accessibility — a link with no text content at all!

The simplest fix is to use an `aria-label` attribute to give the link an accessible name. Slightly better would be to use actual text content inside the `<a></a>`, and then hide it offscreen, but you'd have to dig into the SCSS to figure out what classes would do that.

But using the full image alt text as the link text is sub-optimal, since the purpose of clicking the link is to display the image (and therefore its full alt text). And equally important, it leaves the link label-less if there is no alt text in the database.

I'd recommend using something like "Image 1", "Image 2", "Image 3". Boring, but clear. Ideally, "Image" would be easily localizable using one of the text input configs.  (Same for "left" and "right" on the buttons, while we're at it.)

And since Mustache doesn't provide loop index variables, numbering the images would mean adding an `index` variable in the map created by the [`carouselImages` getter](https://github.com/Shopify/buy-button-js/blob/99d186ff641ea993401431c938b140126769ba5f/src/components/product.js#L215):

```js
  get carouselImages() {
    return this.model.images.map((image, index) => {
      return {
        index,
        id: image.id,
        src: image.src,
        carouselSrc: this.props.client.image.helpers.imageForSize(image, {maxWidth: 100, maxHeight: 100}),
        isSelected: image.id === this.currentImage.id,
      };
    });
  }
```

Anyway, that's a lot of different ideas. The bare minimum fix is to use `aria-label` instead of `alt`:

```suggestion
                        <a data-element="product.carouselitem" aria-label="{{altText}}" href="{{src}}" class="{{data.classes.product.carouselItem}} {{#isSelected}} {{data.classes.product.carouselItemSelected}} {{/isSelected}}" data-image-id="{{id}}" style="background-image: url({{carouselSrc}})"></a>
```

Reply:
@AmeliaBR Thank you for this review. I think I see what you mean with regards to your points on:
1. the decision to choose `<a>` over `<img>` to render the image. Using an `<img>` for the carousel images seems more intuitive.
2. the tradeoffs for using the alt text as the link text -- if alt text is available add it and hide it with SCSS but if alt text is missing then the link text would be blank, which is "sub-optimal".
3. "...using something like 'Image 1', 'Image 2', 'Image 3'" -- this is an interesting solution. I suppose if we were to insert `Image ${index}` as the link text then we would have to use SCSS to hide this text. However at the same time we would not need to worry about blank link text. 
4. "The bare minimum fix is to use aria-label instead of alt" -- Agreed 👍.

And in honor of 'keeping it simple' I'd like to go with option `4`. I'll make those changes now.

Reply:
@AmeliaBR Also what do you mean by:
> But the original code is really, really bad accessibility...

Reply:
The really bad accessibility was having links with absolutely zero text content or aria-label. Using background images instead of `<img>` is minor compared to that.

Reply:
Ok I think I understand. I am relatively new to the [Accessibility API](https://www.w3.org/WAI/PF/aria-implementation/#def_accessibility_api) so I think I got a little confused by the technical intent of the word, `accessibility`. An `aria-label` will make the `<a>` 'accessible' -- where as an `<a>` without text content or an aria-label makes the element unaccessible.